onnxruntime/core/session/abi_session_options.cc

```diff
@@ -184,3 +184,21 @@ ORT_API_STATUS_IMPL(OrtApis::AddSessionConfigEntry, _Inout_ OrtSessionOptions* o
                     _In_z_ const char* config_key, _In_z_ const char* config_value) {
   return onnxruntime::ToOrtStatus(options->value.AddConfigEntry(config_key, config_value));
 }
+
+ORT_API_STATUS_IMPL(OrtApis::AddInitializer, _Inout_ OrtSessionOptions* options, _In_z_ const char* name,
+                    _In_ const OrtValue* val) {
+  if (name == nullptr) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "Received nullptr for name.");
+  }
+
+  if (val == nullptr) {
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "Received nullptr for OrtValue.");
+  }
+
+  auto rc = options->value.initializers_to_share_map.emplace(name, val);
+  if (!rc.second) {
```

Comment:
Given these are two pointers, what's the upside to using 'emplace' vs. 'insert'?